tools/java/org/unicode/cldr/util/DtdData.java

```diff
@@ -1434,12 +1438,23 @@ public int compare(String element, String attribute, String value1, String value
     };
 
     public boolean hasValue(String elementName) {
-        return nameToElement.get(elementName).type == ElementType.PCDATA;
+        // TODO
+        try {
+            return nameToElement.get(elementName).type == ElementType.PCDATA;
+        } catch (Exception e) {
+            System.out.println("Exception in hasValue: " + elementName + "\t" + e.getMessage());
+            return false;
+        }
     }
 
     public boolean isMetadata(XPathParts pathPlain) {
         for (String s : pathPlain.getElements()) {
             Element e = getElementFromName().get(s);
+            if (e == null) {
```

Comment:
Same here. The code should protect against ill-formed input, so L1458 should be:

            if (e != null && e.elementStatus == ElementStatus.metadata) {

But if you need debugging code that could be under a flag.

Reply:
or here.

Reply:
Again, I suspect the bug is elsewhere and this was a work-around. I'll keep working to figure it out.

Reply:
Can't repro anymore. Again I suspect this was the same as the mysterious transient "grammaticalState" bug. I'll revise as you suggested.